src/coreclr/nativeaot/Runtime/thread.h

```diff
@@ -162,6 +166,11 @@ class Thread : private RuntimeThreadLocals
                                                     // On Unix this is an optimization to not queue up more signals when one is
                                                     // still being processed.
     };
+#ifdef HOST_WASM
+    PTR_VOID GetShadowStackTop();
+    void SetShadowStackBottom(PTR_VOID pShadowStack);
+    void SetShadowStackTop(PTR_VOID pShadowStack);
+#endif
```

Comment:
I think it would make sense to move these declarations a bit further down (e. g. after the group `InlinePInvokeReturn` is in).

Also, I would get rid of the DACy `PTR_VOID`s and just use `void*`. We're not going to have a DAC for WASM (or NAOT for that matter).